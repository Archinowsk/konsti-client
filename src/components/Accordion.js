/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

type Props = {
  text: string,
  title: string,
}

type State = {
  open: boolean,
}

const Accordian = (props: Props, state: State) => {
  const [open, setOpen] = React.useState(false)
  const { t } = useTranslation()

  const onClick = () => {
    setOpen(!open)
  }

  return (
    <div className='accordian'>
      {open && (
        <React.Fragment>
          <button onClick={() => onClick()}>
            <FontAwesomeIcon className='accordian-icon' icon='angle-up' />
            <span>
              {t('hide')} {t(`${props.title}`)}
            </span>
          </button>
          <h3>{t(`${props.title}`)}</h3>
          <p>{t(`${props.text}`)}</p>
        </React.Fragment>
      )}
      {!open && (
        <div>
          <button onClick={() => onClick()}>
            <FontAwesomeIcon className='accordian-icon' icon='angle-down' />
            <span>
              {t('show')} {t(`${props.title}`)}
            </span>
          </button>
        </div>
      )}
    </div>
  )
}

export default Accordian
