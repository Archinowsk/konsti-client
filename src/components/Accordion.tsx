import React, { FC, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export interface Props {
  text: string;
  title: string;
  buttonText: string;
}

export const Accordion: FC<Props> = (props: Props): ReactElement => {
  const { text, title, buttonText } = props;

  const [open, setOpen] = React.useState<boolean>(false);

  const { t } = useTranslation();

  const onClick = () => {
    setOpen(!open);
  };

  const splitTextRows = (text: string) => {
    const rows = t(text).split('\n');
    return rows.map(row => <p key={row}>{row}</p>);
  };

  return (
    <div className='accordion'>
      {open && (
        <>
          <button className='accordion-toggle' onClick={() => onClick()}>
            <FontAwesomeIcon className='accordion-icon' icon='angle-up' />
            {t(`${buttonText}`)}
          </button>
          <div className='accordion-content'>
            <h3>{t(`${title}`)}</h3>
            <div>{splitTextRows(text)}</div>
          </div>
        </>
      )}
      {!open && (
        <>
          <button className='accordion-toggle' onClick={() => onClick()}>
            <FontAwesomeIcon className='accordion-icon' icon='angle-down' />
            <span>{t(`${buttonText}`)}</span>
          </button>
        </>
      )}
    </div>
  );
};
