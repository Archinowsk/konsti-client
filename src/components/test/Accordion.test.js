/* @flow */
import React from 'react'
import { shallow } from 'enzyme'
import Accordion from '../Accordion'

describe('Loading', () => {
  it('should render correctly', () => {
    const component = shallow(<Accordion />)
    expect(component).toMatchSnapshot()
  })
})
