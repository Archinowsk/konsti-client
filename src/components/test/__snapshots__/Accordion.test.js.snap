// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  className="accordian"
>
  <div>
    <button
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="accordian-icon"
        fixedWidth={false}
        flip={null}
        icon="angle-down"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span>
        show
         
        undefined
      </span>
    </button>
  </div>
</div>
`;
